### PR TITLE
add initial TPV overwrite search

### PR DESCRIPTION
**Pull request by bgruening**
This PR goes hand in hand with https://github.com/galaxyproject/tpv-shared-database/pull/3

### CODE REVIEW

File: tpv_overwrite.yml [hunk 1]
```diff
@@ -0,0 +1,840 @@
...
+      reject:
+      - pulsar
+    ## TODO this is to investigated
+    params:
+      requirements: GalaxyTraining == false
+      rank: GalaxyGroup == "metadata"
+    context:
+      params:
+        requirements: GalaxyTraining == false
+        rank: GalaxyGroup == "metadata"
```
Comment (nuwang) [inline tpv_overwrite.yml:38]:
```suggestion
```

File: tpv_overwrite.yml [hunk 2]
```diff
@@ -0,0 +1,840 @@
...
+      force_destination_id: true
+  interactive_tool_climate_notebook:
+    mem: 4
+    env:
+      HOME: /home/jovyan
+    scheduling:
+      require:
+      - condor_docker_ie_interactive
+    context:
+      force_destination_id: true
```
Comment (nuwang) [inline tpv_overwrite.yml:67]: `force_destination_id` appears to be a sorting hat specific param. I think we discussed adding support for directly specifying/templating destination name in TPV. Something like:
```
params:
   destination: my-dyn-dest-with-{cores}-{mem}-runner
```
So this function in sorting hat can be simulated or customised: https://github.com/usegalaxy-eu/sorting-hat/blob/0b0758a1b8b72bc0ea5ae198ad2949f4aa16b586/sorting_hat.py#L186
I'll add a test for this.

File: tpv_overwrite.yml [hunk 3]
```diff
@@ -0,0 +1,840 @@
...
+  toolshed.g2.bx.psu.edu/repos/iuc/lumpy_prep/lumpy_prep/.*:
+    mem: 8
+    scheduling:
+      require:
+      - condor_singularity_with_conda
+
+  toolshed.g2.bx.psu.edu/repos/bgruening/pharmcat/pharmcat/.*:
+    mem: 4
+    env:
+      _JAVA_OPTIONS: -Xmx4G -Xms1G
```
Comment (nuwang) [inline tpv_overwrite.yml:307]:
```suggestion
      _JAVA_OPTIONS: -Xmx{int(mem)}G -Xms1G
```

File: tpv_overwrite.yml [hunk 4]
```diff
@@ -0,0 +1,840 @@
...
+    scheduling:
+      require:
+      - docker
+
+  toolshed.g2.bx.psu.edu/repos/iuc/vardict_java/vardict_java/.*:
+    cores: 2
+    mem: 128
+    env:
+      _JAVA_OPTIONS: -Xmx128G -Xms1G
```
Comment (nuwang) [inline tpv_overwrite.yml:316]:
```suggestion
      _JAVA_OPTIONS: -Xmx{int(mem)}G -Xms1G
```

File: tpv_overwrite.yml [hunk 5]
```diff
@@ -0,0 +1,840 @@
...
+    gpus: 1
+    scheduling:
+      require:
+      - remote_cluster_mq_uk01
+
+  toolshed.g2.bx.psu.edu/repos/galaxyp/msgfplus/msgfplus/.*:
+    cores: 10
+    mem: 12
+    env:
+      _JAVA_OPTIONS: -XX:MaxPermSize=2G -Xmx6G -Xms1G -Djava.io.tmpdir=/data/2/galaxy_db/tmp -Duser.home=/data/2/galaxy_db/tmp
```
Comment (nuwang) [inline tpv_overwrite.yml:387]: Should the max heap size `Xmx` match mem here too?

File: tpv_overwrite.yml [hunk 6]
```diff
@@ -0,0 +1,840 @@
...
+
+
+  .*_test_permissions_2.*:
+    rules:
+    
+    not user or 'Alphafold' not in [role.name for role in user.all_roles() if not role.deleted]
+    
+    
+      - if: |
+          not user or user.email != "a@example.org" or user not in roles.role-b
```
Comment (nuwang) [inline tpv_overwrite.yml:708]:
```suggestion
    rules:
      - if: |
            not user or 'Alphafold' not in [role.name for role in user.all_roles() if not role.deleted]
```

File: tpv_overwrite.yml [hunk 7]
```diff
@@ -0,0 +1,840 @@
...
+
+
+
+    context:
+      permissions:
+        allow:
+          users:
+          - a@example.com
+          roles:
+          - role-b
```
Comment (nuwang) [inline tpv_overwrite.yml:722]: Permissions will need to be implemented through tag matching as in this example: https://total-perspective-vortex.readthedocs.io/en/latest/topics/tpv_by_example.html#user-and-role-handling

```suggestion
roles:
  role-b:
    scheduling:
           accept:
             - authorize_dangerous_tool
```

**Comment by kysrpex** [comment]
I assume this can be closed?

**Pull request closed without merging**